koans/ClassesKoans.php

```diff
@@ -0,0 +1,221 @@
+<?php
+namespace PhpKoans;
+
+use PHPUnit\Framework\TestCase;
+
+defined('__') or define('__', null);
+
+class ClassesKoans extends TestCase
+{
+    /**
+     * @testdox Classes can be defined using the `class` keyword and you can create multiple instances of a class (objects)
+     */
+    public function testClassDefinition()
+    {
+        $exampleClass = new ExampleClass();
+
+        $this->assertTrue(__, class_exists('ExampleClass'));
+    }
+
+    /**
+     * @testdox Properties can be defined in a class
+     */
+    public function testClassProperties()
+    {
+        $exampleClass = new ExampleClass();
+
+        $this->assertEquals(__, $exampleClass->exampleProperty);
+    }
+
+    /**
+     * @testdox Methods can be defined in a class
+     */
+    public function testClassMethods()
+    {
+        $exampleClass = new ExampleClass();
+
+        $this->assertEquals(__, $exampleClass->exampleMethod());
+    }
+
+    /**
+     * @testdox Classes can have constructors
+     */
+    public function testClassConstructor()
+    {
+        $car = new Car(__, __);
+
+        $this->assertEquals('blue', $car->color);
+        $this->assertEquals('Ferrari', $car->getBrand());
+    }
+
+    /**
+     * @testdox Classes can have getters and setters methods to access and set value to properties
+     */
+    public function testGettersAndSettersMethods()
+    {
+        $car = new Car('yellow', 'Aston Martin');
+
+        $car->setColor('red');
+
+        $this->assertEquals(__, $car->color);
+        $this->assertEquals(__, $car->getBrand());
+    }
+
+    /**
+     * @testdox Classes can have static properties and methods
+     */
+    public function testStaticPropertiesAndMethods()
+    {
+        Car::$counter++;
+
+        $this->assertEquals(__, Car::$counter);
+        $this->assertEquals(__, Car::getCount());
+    }
+
+    /**
+     * @testdox Private methods and properties cannot be accessed directly from outside the class
+     */
+    public function testPrivateAccess()
+    {
+        $car = new Car('yellow', 'Mercedes');
+
+        $this->expectException(\Error::class);
+        $this->expectExceptionMessage('Fix me');
+        $car->secretProperty;
+
+        $this->expectException(\Error::class);
+        $this->expectExceptionMessage('Fix me');
+        $car->secretMethod();
+    }
+
+    /**
+     * @testdox Protected methods and properties cannot be accessed directly from outside the class
+     */
+    public function testProtectedAccess()
+    {
+        $car = new Car('yellow', 'Mercedes');
+
+        $this->assertEquals('This is protected', $car->protectedProperty);
+    }
+
+    /**
+     * @testdox Child classes can inherit properties and methods from their parent class
+     */
+    public function testInheritance()
```

Comment:
This test is very similar with four above this

Reply:
If it is to make it clear how childs work, then...ok, no?